### PR TITLE
feat: :sparkles: Added basic GTFS domain specific blocks to stdLib

### PR DESCRIPTION
**Pull request by rhazn**
When I was describing Jayvee at TU Berlin, I felt like we are missing a stronger, domain-specific example.

This PR uses the new composite block capabilities and some of the GTFS domain knowledge we developed to start a "domain extension" for mobility data to the stdLib. Using this extension, we can shorten the GTFS example quite a bit from 358 lines to 91 (and the result should be better because we are using domain specific value types).

Further progress could be made with configurable blocks (to reduce the boilerplate when creating SQLite sinks that only differ in a table name) and better ways to describe DAGs (to reduce the initial bloat with creating many different lines instead of one graph).

Let me know what you think of this in general, I'd like to create something similar for SWC (materials science domain). We can slowly move value types that are more generic than the domain they are initially defined in to a higher level (e.g. I anticipate the Latitude and Longitude value types are not mobility specific).

### CODE REVIEW

File: libs/language-server/src/stdlib/domain/mobility/GTFSAgencyInterpreter.jv [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    block AgencyFilePicker oftype FilePicker {
+		path: "/agency.txt";
+	}
+
+    block AgencyTextFileInterpreter oftype TextFileInterpreter { }
+    block AgencyCSVInterpreter oftype CSVInterpreter { }
+
+    block AgencyTableInterpreter oftype TableInterpreter {
+		header: true;
+		columns:[
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSAgencyInterpreter.jv:30]:
```suggestion
		columns: [
```

File: libs/language-server/src/stdlib/domain/mobility/GTFSAgencyInterpreter.jv [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+			"agency_id" oftype text, //Conditional columns are considered required for now
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSAgencyInterpreter.jv:31]:
```suggestion
			"agency_id" oftype text, // Conditional columns are considered required for now
```

File: libs/language-server/src/stdlib/domain/mobility/GTFSFareAttributesInterpreter.jv [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    block FareAttributesFilePicker oftype FilePicker {
+		path: "/fare_attributes.txt";
+	}
+
+    block FareAttributesTextFileInterpreter oftype TextFileInterpreter { }
+    block FareAttributesCSVInterpreter oftype CSVInterpreter { }
+
+    block FareAttributesTableInterpreter oftype TableInterpreter {
+		header: true;
+		columns:[
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSFareAttributesInterpreter.jv:30]:
```suggestion
		columns: [
```

File: libs/language-server/src/stdlib/domain/mobility/GTFSFareRulesInterpreter.jv [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    block FareRulesFilePicker oftype FilePicker {
+		path: "/fare_rules.txt";
+	}
+
+    block FareRulesTextFileInterpreter oftype TextFileInterpreter { }
+    block FareRulesCSVInterpreter oftype CSVInterpreter { }
+
+    block FareRulesTableInterpreter oftype TableInterpreter {
+		header: true;
+		columns:[
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSFareRulesInterpreter.jv:30]:
```suggestion
		columns: [
```

File: libs/language-server/src/stdlib/domain/mobility/GTFSFrequenciesInterpreter.jv [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    block FrequenciesFilePicker oftype FilePicker {
+		path: "/frequencies.txt";
+	}
+
+    block FrequenciesTextFileInterpreter oftype TextFileInterpreter { }
+    block FrequenciesCSVInterpreter oftype CSVInterpreter { }
+
+    block FrequenciesTableInterpreter oftype TableInterpreter {
+		header: true;
+		columns:[
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSFrequenciesInterpreter.jv:30]:
```suggestion
		columns: [
```

File: libs/language-server/src/stdlib/domain/mobility/GTFSStopTimesInterpreter.jv [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    block StopTimesFilePicker oftype FilePicker {
+		path: "/stop_times.txt";
+	}
+
+    block StopTimesTextFileInterpreter oftype TextFileInterpreter { }
+    block StopTimesCSVInterpreter oftype CSVInterpreter { }
+
+    block StopTimesTableInterpreter oftype TableInterpreter {
+		header: true;
+		columns:[
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSStopTimesInterpreter.jv:30]:
```suggestion
		columns: [
```

File: libs/language-server/src/stdlib/domain/mobility/GTFSStopsInterpreter.jv [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    block StopsFilePicker oftype FilePicker {
+		path: "/stops.txt";
+	}
+
+    block StopsTextFileInterpreter oftype TextFileInterpreter { }
+    block StopsCSVInterpreter oftype CSVInterpreter { }
+
+    block StopsTableInterpreter oftype TableInterpreter {
+		header: true;
+		columns:[
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSStopsInterpreter.jv:30]:
```suggestion
		columns: [
```

File: libs/language-server/src/stdlib/domain/mobility/GTFSTripsInterpreter.jv [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    block TripsFilePicker oftype FilePicker {
+		path: "/trips.txt";
+	}
+
+    block TripsTextFileInterpreter oftype TextFileInterpreter { }
+    block TripsCSVInterpreter oftype CSVInterpreter { }
+
+    block TripsTableInterpreter oftype TableInterpreter {
+		header: true;
+		columns:[
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSTripsInterpreter.jv:30]:
```suggestion
		columns: [
```

File: libs/language-server/src/stdlib/domain/mobility/GTFSValueTypes.jv [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+valuetype GTFSNonNegativeInteger oftype integer {
+    constraints: [NonNegativeNumber];
+}
+
+constraint URLIncludingSchema on text: value matches /^(http)s?(:\/\/)/;
+valuetype GTFSUrl oftype text {
+    constraints: [URLIncludingSchema];
+}
```
Comment (joluj) [inline libs/language-server/src/stdlib/domain/mobility/GTFSValueTypes.jv:99]: Add a new line
> “lines” not ending in a newline character aren't considered actual lines.

(https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

**Review by joluj — APPROVED** [review]
I didn't check it in detail, but makes sense to me.

Spaces and Tabs are mixed, reformatting the code would be nice. Creating a formatter is still blocked by https://github.com/eclipse-langium/langium/issues/1351.

File: libs/language-server/src/stdlib/domain/mobility/GTFSValueTypes.jv [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+ * Phone number - A phone number.
+ * Time - Time in the HH:MM:SS format (H:MM:SS is also accepted). The time is measured from "noon minus 12h" of the service day (effectively midnight except for days on which daylight savings time changes occur). For times occurring after midnight on the service day, enter the time as a value greater than 24:00:00 in HH:MM:SS.
+ *      Example: 14:30:00 for 2:30PM or 25:35:00 for 1:35AM on the next day.
+ * Text - A string of UTF-8 characters, which is aimed to be displayed and which must therefore be human readable.
+ * Timezone - TZ timezone from the https://www.iana.org/time-zones. Timezone names never contain the space character but may contain an underscore. Refer to http://en.wikipedia.org/wiki/List_of_tz_zones for a list of valid values.
+ *      Example: Asia/Tokyo, America/Los_Angeles or Africa/Cairo.
+ * URL - A fully qualified URL that includes http:// or https://, and any special characters in the URL must be correctly escaped. See the following http://www.w3.org/Addressing/URL/4_URI_Recommentations.html for a description of how to create fully qualified URL values.
+ * 
+ * https://gtfs.org/schedule/reference/#field-types
+*/
```
Comment (georg-schwarz) [inline libs/language-server/src/stdlib/domain/mobility/GTFSValueTypes.jv:37]: I think those comments should be directly with the value type

**Review by georg-schwarz — APPROVED** [review]
Love it!

File: libs/language-server/src/stdlib/domain/mobility/GTFSValueTypes.jv (hunk 9 above)
Comment (rhazn) [reply libs/language-server/src/stdlib/domain/mobility/GTFSValueTypes.jv:99]: Wow I've learned something today, thanks.

**Pull request merged**